src/field_theory/galois.lean

```diff
@@ -306,6 +306,86 @@ begin
   { exact polynomial.map_ne_zero (minimal_polynomial.ne_zero h2) }
 end
 
+lemma of_fixed_field_eq_bot [finite_dimensional F E]
+  (h : intermediate_field.fixed_field (⊤ : subgroup (E ≃ₐ[F] E)) = ⊥) : is_galois F E :=
+begin
+  rw [←is_galois_iff_is_galois_bot, ←h],
+  exact is_galois.of_fixed_field E (⊤ : subgroup (E ≃ₐ[F] E)),
+end
+
+lemma of_card_aut_eq_findim [finite_dimensional F E]
+  (h : fintype.card (E ≃ₐ[F] E) = findim F E) : is_galois F E :=
+begin
+  apply of_fixed_field_eq_bot,
+  have p : 0 < findim (intermediate_field.fixed_field (⊤ : subgroup (E ≃ₐ[F] E))) E := findim_pos,
+  rw [←intermediate_field.findim_eq_one_iff, ←mul_left_inj' (ne_of_lt p).symm, findim_mul_findim,
+      ←h, one_mul, intermediate_field.findim_fixed_field_eq_card],
+  exact fintype.card_congr { to_fun := λ g, ⟨g, subgroup.mem_top g⟩, inv_fun := coe,
+    left_inv := λ g, rfl, right_inv := λ _, by { ext, refl } },
+end
+
+variables {F} {E} {p : polynomial F}
+
+lemma of_separable_splitting_field_aux [hFE : finite_dimensional F E]
+  [sp : p.is_splitting_field F E] (hp : p.separable) (K : intermediate_field F E) {x : E}
+  (hx : x ∈ (p.map (algebra_map F E)).roots) :
+fintype.card ((↑K⟮x⟯ : intermediate_field F E) →ₐ[F] E) =
+  fintype.card (K →ₐ[F] E) * findim K K⟮x⟯ :=
+begin
+  have h : is_integral K x := is_integral_of_is_scalar_tower x (is_integral_of_noetherian hFE x),
+  have h1 : p ≠ 0 := λ hp, by rwa [hp, polynomial.map_zero, polynomial.roots_zero] at hx,
+  have h2 : (minimal_polynomial h) ∣ p.map (algebra_map F K),
+  { apply minimal_polynomial.dvd,
+    rw [polynomial.aeval_def, polynomial.eval₂_map, ←polynomial.eval_map],
+    exact (polynomial.mem_roots (polynomial.map_ne_zero h1)).mp hx },
+  have key_equiv : ((↑K⟮x⟯ : intermediate_field F E) →ₐ[F] E) ≃ Σ (f : K →ₐ[F] E),
```

Comment:
```suggestion
  let key_equiv : ((↑K⟮x⟯ : intermediate_field F E) →ₐ[F] E) ≃ Σ (f : K →ₐ[F] E),
```
I'm somewhat surprised that this worked with `have`, because it immediately forgets the definition of the equiv.

Reply:
I think that the reason why is that I just needed existence of an equiv to conclude that two things had the same cardinality.